python/xorbits/_mars/dataframe/datasource/read_csv.py

```diff
@@ -84,14 +84,28 @@ def _find_chunk_start_end(f: Any, offset: int, size: int, is_first: bool):
     if is_first:
         start = offset
     else:
-        f.readline()
+        _read_csv_to_line_end(f)
         start = f.tell()
     f.seek(offset + size)
-    f.readline()
+    _read_csv_to_line_end(f)
     end = f.tell()
     return start, end
 
 
+def _read_csv_to_line_end(f: Any) -> None:
+    f.readline()
+    file_ptr = f.tell()
+    f.seek(0)
+    while (f.read(file_ptr)).count(b'"') % 2 == 1:  # pragma: no cover
```

Comment:
It's wired to `read(file_ptr)`. Please refer to the file doc: https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=read#fsspec.spec.AbstractBufferedFile.read

Reply:
Maybe you can define a max len for a csv line. Then, we only ensure that if some newlines inside a csv line can be handled. @aresnow1 What do you think?